src/nvim/syntax.c

```diff
@@ -5791,7 +5783,7 @@ int syn_get_id(win_T *wp, long lnum, colnr_T col, int trans, bool *spellp, int k
   // When the position is not after the current position and in the same
   // line of the same window with the same buffer, need to restart parsing.
   if (wp != syn_win || wp->w_buffer != syn_buf || lnum != current_lnum || col < current_col) {
-    syntax_start(wp, lnum);
+    syntax_start(wp, (linenr_T)lnum);
```

Comment:
This `lnum` argument is only passed a `linenr_T`, so it can be changed to a `linenr_T`.